engine/components-api/src/main/java/com/cloud/template/TemplateManager.java

```diff
@@ -51,6 +51,11 @@ public interface TemplateManager {
     static final ConfigKey<Integer> TemplatePreloaderPoolSize = new ConfigKey<Integer>("Advanced", Integer.class, TemplatePreloaderPoolSizeCK, "8",
             "Size of the TemplateManager threadpool", false, ConfigKey.Scope.Global);
 
+    ConfigKey<Boolean> ValidateUrlIsResolvableBeforeRegisteringTemplate = new ConfigKey<>("Advanced", Boolean.class,
+            "validate.url.is.resolvable.before.registering.template", "true", "Indicates whether CloudStack "
+            + "will validate if the provided URL is resolvable during the register of templates/ISOs before persisting them in the database.",
+            true);
```

Comment:
@winterhazel , this is now a global setting. Does it make sense to norrow the scope? 

i.e. allow it only in specific zones/clusters, like dedicated ones?

Reply:
@DaanHoogland I think it would make sense to have it management server scoped, as its use-case comes from a management server specific configuration; making it zone/cluster scoped does not make sense as the same management server can manage multiple zones/clusters. 

However, even though the `ManagementServer` scope type exists, its implementation has not been made yet. I think we could implement it in a separate PR and change this setting in the future, though.